### PR TITLE
Convert Danishbits to use their api

### PR DESCRIPTION
**Pull request by duramato**
- [X] PR is based on the DEVELOP branch
- [X] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/providers/torrent/json/danishbits.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                        log.debug("Discarding torrent because it doesn't meet the"
+                                  " minimum seeders: {0}. Seeders: {1}",
+                                  title, seeders)
+                    continue
+
+                freeleech = row.get('freeleech')
+                if self.freeleech and not freeleech:
+                    continue
+
+                size = try_int(row.get('size')) or -1
```
Comment (p0psicles) [inline medusa/providers/torrent/json/danishbits.py]: What do you need the try_int for?

Comment (duramato) [inline medusa/providers/torrent/json/danishbits.py]: Size isn't working, i don't why

File: medusa/providers/torrent/json/danishbits.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        self.username = None
+        self.passkey = None
+
+        # URLs
+        self.url = 'https://danishbits.org'
+        self.urls = {
+            'search': urljoin(self.url, 'couchpotato.php'),
+        }
+
+        # Proper Strings
```
Comment (medariox) [inline medusa/providers/torrent/json/danishbits.py]: No proper strings?

File: medusa/providers/torrent/json/danishbits.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        results = []
```
Comment (medariox) [inline medusa/providers/torrent/json/danishbits.py]: Double results.

File: medusa/providers/torrent/json/danishbits.py (hunk 2 above)
Comment (duramato) [inline medusa/providers/torrent/json/danishbits.py]: Means it uses the ones from generic. Which you reminded me to check if they work on the API.

Comment (fernandog) [inline medusa/providers/torrent/json/danishbits.py]: @duramato provider allows an OR search?

its the difference of using:
self.proper_strings = ['PROPER|REPACK|REAL|RERIP']
or 
self.proper_strings = ['PROPER', 'REPACK', 'REAL', 'RERIP']

Comment (duramato) [inline medusa/providers/torrent/json/danishbits.py]: https://github.com/pymedusa/Medusa/blob/master/medusa/providers/generic_provider.py#L92

No need to overwride it. Waiting on for the site to support it.

Comment (fernandog) [inline medusa/providers/torrent/json/danishbits.py]: we only need to override if they doesn't support OR search

File: medusa/providers/torrent/json/danishbits.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                    continue
+
+                freeleech = row.get('freeleech')
+                if self.freeleech and not freeleech:
+                    continue
+
+                torrent_size = '{0} MB'.format(row.get('size', -1))
+                size = convert_size(torrent_size) or -1
+
+                pubdate = row.get('publish_date')
```
Comment (p0psicles) [inline medusa/providers/torrent/json/danishbits.py]: We don't need any date parsing here?

**Comment by p0psicles** [comment]
I'll try to do some testing this weekend. And then this should be merged.

File: medusa/providers/torrent/json/danishbits.py (hunk 4 above)
Comment (duramato) [inline medusa/providers/torrent/json/danishbits.py]: It's in ISO format

Comment (p0psicles) [inline medusa/providers/torrent/json/danishbits.py]: Cool. Can you look into the proper? Then we can get it merged.

**Comment by p0psicles** [comment]
@duramato, it's ready now?

**Comment by duramato** [comment]
Daily isn't working, i only did a hacky way for it to work by searching for `x264`. RSS needs to be added to the api.
Everything else is ready

File: medusa/providers/torrent/json/danishbits.py [hunk 5]
```diff
@@ -63,7 +63,7 @@ def search(self, search_strings, age=0, ep_obj=None):
         search_params = {
             'user': self.username,
             'passkey': self.passkey,
-            'search': '.', # Dummy query for RSS search, needs the search param sent
+            'search': '.',  # Dummy query for RSS search, needs the search param send.
```
Comment (duramato) [inline medusa/providers/torrent/json/danishbits.py]: `send`?

**Pull request merged**